### PR TITLE
Order of keys shouldn't matter when comparing cirq.google.KeyValueExecutableSpec

### PR DESCRIPTION
**Pull request by tonybruguier**
For issue #4734. It was discussed there that the order of the keys should not matter.

### CODE REVIEW

File: cirq-google/cirq_google/workflow/quantum_executable_test.py [hunk 1]
```diff
@@ -187,3 +187,12 @@ def test_quantum_executable_group_serialization(tmpdir):
     cirq.to_json(eg, f'{tmpdir}/eg.json')
     eg_reconstructed = cirq.read_json(f'{tmpdir}/eg.json')
     assert eg == eg_reconstructed
+
+
+def test_equality():
+    d1 = dict.fromkeys(['a', 'b'])
+    d2 = dict.fromkeys(['b', 'a'])
```
Comment (mpharrigan) [inline cirq-google/cirq_google/workflow/quantum_executable_test.py:194]: why don't these dictionaries have values?

Comment (tonybruguier) [reply cirq-google/cirq_google/workflow/quantum_executable_test.py:194]: The example from Tanuj was like this:
https://github.com/quantumlib/Cirq/issues/4734#issuecomment-995168589

But I changed the dictionaries to have keys. I also verified that the unit test still fails if I don't override the __eq__ operator.

**Comment by tonybruguier** [comment]
Thanks, @mpharrigan and PTAL when you have the time? It's not urgent on my side.

**Comment by mpharrigan** [comment]
Can you add a section to the docstring about the new equality semantics of KeyValueSpec since we're overwriting the auto-generated dataclass equality method? And add a test for the new equality semantics that constructs the kvspec using tuples and the ordinary constructor?

**Comment by tonybruguier** [comment]
Like so?

**Review by mpharrigan — APPROVED** [review]

**Comment by tonybruguier** [comment]
I think this could be ready to merge. But happy to work more on it if there are further comments, of course.

**Pull request merged**